core/client/src/main/java/alluxio/client/block/BlockStoreContext.java

```diff
@@ -313,11 +227,50 @@ public static void releaseNettyChannel(InetSocketAddress address, Channel channe
   }
 
   /**
+   * Acquires a block worker thrift client from the block worker thrift client pools.
+   *
+   * @param address the address of the block worker
+   * @return the block worker thrift client
+   * @throws IOException if it fails to connect to remote worker
+   */
+  public static BlockWorkerClientService.Client acquireBlockWorkerThriftClient(
+      final InetSocketAddress address) throws IOException {
+    if (!BLOCK_WORKER_THRIFT_CLIENT_POOL.containsKey(address)) {
+      BlockWorkerThriftClientPool pool = new BlockWorkerThriftClientPool(address,
+          Configuration.getInt(PropertyKey.USER_BLOCK_WORKER_CLIENT_POOL_SIZE_MAX),
+          Configuration.getInt(PropertyKey.USER_BLOCK_WORKER_CLIENT_POOL_GC_THRESHOLD_SECS));
+      if (BLOCK_WORKER_THRIFT_CLIENT_POOL.putIfAbsent(address, pool) != null) {
+        pool.close();
+      }
+    }
+    try {
+      return BLOCK_WORKER_THRIFT_CLIENT_POOL.get(address).acquire();
+    } catch (IOException e) {
+      LOG.error(String.format("Failed to connect to block worker %s.", address), e);
+      throw e;
+    }
+  }
+
+  /**
+   * Releases the block worker thrift client to the pool.
+   *
+   * @param address the network address of the block worker thrift client
+   * @param client the block worker thrift client
+   */
+  public static void releaseBlockWorkerThriftClient(InetSocketAddress address,
+      BlockWorkerClientService.Client client) {
+    Preconditions.checkArgument(BLOCK_WORKER_THRIFT_CLIENT_POOL.containsKey(address));
+    BLOCK_WORKER_THRIFT_CLIENT_POOL.get(address).release(client);
```

Comment:
ack. so this property is ensured by the fact that the map is always growing. As said previously, better to javadoc this assumption in the map definition (in case we change the behavior of the map but forget to update or protect code here)